src/devices/src/virtio/block/event_handler.rs

```diff
@@ -27,25 +58,120 @@ impl Subscriber for Block {
         let supported_events = EventSet::IN;
         if !supported_events.contains(event_set) {
             warn!(
-                "Received unknown event: {:?} from source: {:?}",
+                "Block: Received unknown event: {:?} from source: {:?}",
                 event_set, source
             );
             return;
         }
 
-        // Looks better than C style if/else if/else.
-        match source {
-            _ if queue_evt == source => self.process_queue_event(),
-            _ if rate_limiter_evt == source => self.process_rate_limiter_event(),
-            _ => warn!("Spurious event received: {:?}", source),
-        }
+        match self.device_state {
+            DeviceState::Activated(_) => {
+                let queue_evt = self.queue_evts[0].as_raw_fd();
+                let rate_limiter_evt = self.rate_limiter.as_raw_fd();
+                let activate_fd = self.activate_evt.as_raw_fd();
+
+                // Looks better than C style if/else if/else.
+                match source {
+                    _ if queue_evt == source => self.process_queue_event(),
+                    _ if rate_limiter_evt == source => self.process_rate_limiter_event(),
+                    _ if activate_fd == source => self.process_activate_event(evmgr),
+                    _ => warn!("Block: Spurious event received: {:?}", source),
+                }
+            }
+            DeviceState::Inactive => warn!(
+                "Block: The device is not yet activated. Spurious event received: {:?}",
+                source
+            ),
+        };
     }
 
     // Returns the rate_limiter and queue event fds.
```

Comment:
```suggestion
    // Returns the activate event fd.
```

Reply:
Could also remove the comment, as it's not essential and requires maintenance with each change of the function's meaning.

Reply:
Done.